src/test-env.ts

```diff
@@ -0,0 +1,119 @@
+import { addressUtils, HDNode, unitsUtils } from '@vechain/sdk-core'
+import { ThorClient } from '@vechain/sdk-network'
+import { genesis } from './constants'
+
+const mnemonic = process.env.MNEMONIC
+const nodeUrls = process.env.NODE_URLS
+const privateKeys = process.env.PRIVATE_KEYS
+const networkType = process.env.NETWORK_TYPE
+
+const NetworkTypes = ['solo', 'default-private', 'testnet', 'mainnet'] as const
+type NetworkType = (typeof NetworkTypes)[number]
+
+const testEnv = {
+    get keys(): string[] {
+        if (privateKeys) {
+            return privateKeys.split(',')
+        }
+
+        if (!mnemonic) {
+            throw new Error('No mnemonic provided')
+        }
+        const _mnemonic = mnemonic
+
+        const hdNode = HDNode.fromMnemonic(_mnemonic.split(' '))
+        const keys: any[] = []
+        for (let i = 0; i < 10; i++) {
+            const child = hdNode.derive(i)
+            keys.push(Buffer.from(child.privateKey!).toString('hex'))
+        }
+
+        return keys
+    },
+    get urls(): string[] {
+        if (nodeUrls) {
+            return nodeUrls.split(',')
+        }
+
+        return ['http://localhost:8669']
+    },
+    get type(): NetworkType {
+        if (networkType) {
+            const type = networkType as NetworkType
+            if (!NetworkTypes.includes(type)) {
+                throw new Error('Invalid network type')
+            }
+            return type
+        }
+
+        return 'default-private'
+    },
+}
+
+const validateEnv = async (): Promise<boolean> => {
+    const genesisIds: string[] = []
+
+    if (nodeUrls) {
+        const urls = nodeUrls.split(',')
+        for (const url of urls) {
+            const client = ThorClient.fromUrl(url)
+            const genesis = await client.blocks.getGenesisBlock()
+            genesisIds.push(genesis!.id)
+        }
+
+        // Ensure all genesis blocks match
+        if (new Set(genesisIds).size !== 1) {
+            console.error('Genesis blocks do not match')
+            return false
+        }
+
+        // Ensure private keys OR mnemonic are provided for testnet
+        if (genesisIds.includes(genesis.test) && !(privateKeys || mnemonic)) {
+            console.error(
+                'Testnet genesis block detected but no private keys provided',
+            )
+            return false
+        }
+
+        // Ensure private keys OR mnemonic are provided for mainnet
+        if (genesisIds.includes(genesis.main) && !(privateKeys || mnemonic)) {
+            console.error(
+                'Mainnet genesis block detected but no private keys provided',
+            )
+            return false
+        }
+    }
+
+    // const accounts = testEnv.keys.map((key) => {
+    //     if (key.startsWith('0x')) {
+    //         key = key.slice(2)
+    //     }
+    //     return addressUtils.fromPrivateKey(
+    //         Uint8Array.from(Buffer.from(key, 'hex')),
+    //     )
+    // })
+
+    // TODO: Validate the account balances
```

Comment:
anything we need to do right now with both commented code / todos here ?

Reply:
I think we need to deduct some minimal amount of VET and VTHO for the tests to be able to be ran

Reply:
probably future improvement